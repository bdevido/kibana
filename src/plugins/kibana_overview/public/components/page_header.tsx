/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React, { FC } from 'react';
import { EuiFlexGroup, EuiFlexItem, EuiButtonEmpty } from '@elastic/eui';
import { i18n } from '@kbn/i18n';
import { createAppNavigationHandler } from './app_navigation_handler';
import { FeatureCatalogueEntry } from '../../services';

interface Props {
  title: JSX.Element;
  findDirectoryById: (id: string) => FeatureCatalogueEntry | undefined;
}

export const PageHeader: FC<Props> = ({ title, findDirectoryById }) => {
  const devTools = findDirectoryById('console');
  const stackManagement = findDirectoryById('stack-management');

  return (
    <div className="homPageHeaderContainer">
      <header className="homPageHeader">
        <EuiFlexGroup gutterSize="none">
          <EuiFlexItem className="homPageHeader__titleWrapper">{title}</EuiFlexItem>
          <EuiFlexItem grow={false}>
            <EuiFlexGroup className="homPageHeader__menu" alignItems="flexEnd">
              <EuiFlexItem className="homPageHeader__menuItem">
                <EuiButtonEmpty href="#/tutorial_directory" iconType="plusInCircle">
                  {i18n.translate('home.pageHeader.addDataButtonLabel', {
                    defaultMessage: 'Add data',
                  })}
                </EuiButtonEmpty>
              </EuiFlexItem>
              {stackManagement ? (
                <EuiFlexItem className="homPageHeader__menuItem">
                  <EuiButtonEmpty
                    onClick={createAppNavigationHandler(stackManagement.path)}
                    iconType="gear"
                  >
                    {i18n.translate('home.pageHeader.stackManagementButtonLabel', {
                      defaultMessage: 'Manage',
                    })}
                  </EuiButtonEmpty>
                </EuiFlexItem>
              ) : null}
              {devTools ? (
                <EuiFlexItem className="homPageHeader__menuItem">
                  <EuiButtonEmpty
                    onClick={createAppNavigationHandler(devTools.path)}
                    iconType="wrench"
                  >
                    {i18n.translate('home.pageHeader.devToolsButtonLabel', {
                      defaultMessage: 'Dev tools',
                    })}
                  </EuiButtonEmpty>
                </EuiFlexItem>
              ) : null}
            </EuiFlexGroup>
          </EuiFlexItem>
        </EuiFlexGroup>
      </header>
    </div>
  );
};
