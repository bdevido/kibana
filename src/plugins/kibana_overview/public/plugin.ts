/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import { i18n } from '@kbn/i18n';
import {
  AppMountParameters,
  CoreSetup,
  CoreStart,
  Plugin,
  DEFAULT_APP_CATEGORIES,
} from '../../../../src/core/public';
import {
  KibanaOverviewPluginSetup,
  KibanaOverviewPluginStart,
  AppPluginSetupDependencies,
  AppPluginStartDependencies,
} from './types';
import { PLUGIN_NAME, PLUGIN_PATH } from '../common';
import { setServices } from './kibana_services';

export class KibanaOverviewPlugin
  implements
    Plugin<
      KibanaOverviewPluginSetup,
      KibanaOverviewPluginStart,
      AppPluginSetupDependencies,
      AppPluginStartDependencies
    > {
  public setup(
    core: CoreSetup<AppPluginStartDependencies>,
    { home }: AppPluginSetupDependencies
  ): KibanaOverviewPluginSetup {
    // Register an application into the side navigation menu
    core.application.register({
      category: DEFAULT_APP_CATEGORIES.kibana,
      id: 'kibanaOverview',
      title: PLUGIN_NAME,
      order: 1,
      appRoute: PLUGIN_PATH,
      async mount(params: AppMountParameters) {
        // Load application bundle
        const { renderApp } = await import('./application');
        // Get start services as specified in kibana.json
        const [coreStart, depsStart] = await core.getStartServices();
        setServices({
          addBasePath: core.http.basePath.prepend,
          application: coreStart.application,
          banners: coreStart.overlays.banners,
          chrome: coreStart.chrome,
          docLinks: coreStart.docLinks,
          getBasePath: core.http.basePath.get,
          http: coreStart.http,
          indexPatternService: depsStart.data.indexPatterns,
          savedObjectsClient: coreStart.savedObjects.client,
          toastNotifications: core.notifications.toasts,
          uiSettings: core.uiSettings,
        });

        // Render the application
        return renderApp(coreStart, depsStart as AppPluginStartDependencies, params);
      },
    });

    if (home) {
      home.featureCatalogue.registerSolution({
        id: 'kibana',
        title: i18n.translate('home.kibana.featureCatalogue.title', {
          defaultMessage: 'Kibana',
        }),
        subtitle: i18n.translate('home.kibana.featureCatalogue.subtitle', {
          defaultMessage: 'Visualize & analyze',
        }),
        descriptions: [
          i18n.translate('home.kibana.featureCatalogueDescription1', {
            defaultMessage: 'Analyze data in dashboards.',
          }),
          i18n.translate('home.kibana.featureCatalogueDescription2', {
            defaultMessage: 'Search and find insights.',
          }),
          i18n.translate('home.kibana.featureCatalogueDescription3', {
            defaultMessage: 'Design pixel-perfect reports.',
          }),
          i18n.translate('home.kibana.featureCatalogueDescription4', {
            defaultMessage: 'Plot geographic data.',
          }),
          i18n.translate('home.kibana.featureCatalogueDescription5', {
            defaultMessage: 'Model, predict, and detect.',
          }),
          i18n.translate('home.kibana.featureCatalogueDescription6', {
            defaultMessage: 'Reveal patterns and relationships.',
          }),
        ],
        icon: 'logoKibana',
        path: PLUGIN_PATH,
        order: 400,
      });
    }

    // Return methods that should be available to other plugins
    return {};
  }

  public start(core: CoreStart): KibanaOverviewPluginStart {
    return {};
  }

  public stop() {}
}
