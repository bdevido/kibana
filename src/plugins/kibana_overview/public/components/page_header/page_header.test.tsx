/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React from 'react';
import { PageHeader } from './page_header';
import { shallowWithIntl } from 'test_utils/enzyme_helpers';
import { FeatureCatalogueCategory } from '../../../../../../src/plugins/home/public';

jest.mock('../../app_navigation_handler', () => {
  return {
    createAppNavigationHandler: jest.fn(() => () => {}),
  };
});

beforeEach(() => {
  jest.clearAllMocks();
});

const mockDevTools = {
  category: FeatureCatalogueCategory.ADMIN,
  description: 'Skip cURL and use a JSON interface to work with your data in Console.',
  icon: 'consoleApp',
  id: 'console',
  path: 'path-to-dev-tools',
  showOnHomePage: false,
  title: 'Interact with the Elasticsearch API',
};

const mockManagement = {
  category: FeatureCatalogueCategory.ADMIN,
  description: 'Your center console for managing the Elastic Stack.',
  icon: 'managementApp',
  id: 'stack-management',
  path: 'path-to-management',
  showOnHomePage: false,
  title: 'Stack Management',
};

describe('PageHeader', () => {
  test('render', () => {
    const component = shallowWithIntl(<PageHeader features={[]} />);
    expect(component).toMatchSnapshot();
  });
  test('with dev tools', () => {
    const component = shallowWithIntl(<PageHeader features={[mockDevTools]} />);
    expect(component).toMatchSnapshot();
  });
  test('with management', () => {
    const component = shallowWithIntl(<PageHeader features={[mockManagement]} />);
    expect(component).toMatchSnapshot();
  });
  test('with both apps', () => {
    const component = shallowWithIntl(<PageHeader features={[mockDevTools, mockManagement]} />);
    expect(component).toMatchSnapshot();
  });
});
