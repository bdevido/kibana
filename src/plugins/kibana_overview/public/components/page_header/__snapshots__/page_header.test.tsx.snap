// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader render 1`] = `
<header
  className="kbnOverviewHeader"
>
  <div
    className="kbnOverviewHeader__inner"
  >
    <EuiFlexGroup>
      <EuiFlexItem>
        <EuiFlexGroup
          gutterSize="m"
          responsive={false}
        >
          <EuiFlexItem
            grow={false}
          >
            <EuiIcon
              size="xxl"
              type="logoKibana"
            />
          </EuiFlexItem>
          <EuiFlexItem>
            <EuiTitle
              size="m"
            >
              <h1
                id="kbnOverviewHeader__title"
              >
                <FormattedMessage
                  defaultMessage="Kibana"
                  id="kibana.overview.header.title"
                  values={Object {}}
                />
              </h1>
            </EuiTitle>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiFlexItem>
      <EuiFlexItem
        grow={false}
      >
        <EuiFlexGroup
          className="kbnOverviewHeader__actions"
          responsive={false}
          wrap={true}
        >
          <EuiFlexItem
            className="kbnOverviewHeader__actionItem"
            grow={false}
          >
            <EuiButtonEmpty
              iconType="indexOpen"
              onClick={[Function]}
            >
              Add data
            </EuiButtonEmpty>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiFlexItem>
    </EuiFlexGroup>
  </div>
</header>
`;

exports[`PageHeader with both apps 1`] = `
<header
  className="kbnOverviewHeader"
>
  <div
    className="kbnOverviewHeader__inner"
  >
    <EuiFlexGroup>
      <EuiFlexItem>
        <EuiFlexGroup
          gutterSize="m"
          responsive={false}
        >
          <EuiFlexItem
            grow={false}
          >
            <EuiIcon
              size="xxl"
              type="logoKibana"
            />
          </EuiFlexItem>
          <EuiFlexItem>
            <EuiTitle
              size="m"
            >
              <h1
                id="kbnOverviewHeader__title"
              >
                <FormattedMessage
                  defaultMessage="Kibana"
                  id="kibana.overview.header.title"
                  values={Object {}}
                />
              </h1>
            </EuiTitle>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiFlexItem>
      <EuiFlexItem
        grow={false}
      >
        <EuiFlexGroup
          className="kbnOverviewHeader__actions"
          responsive={false}
          wrap={true}
        >
          <EuiFlexItem
            className="kbnOverviewHeader__actionItem"
            grow={false}
          >
            <EuiButtonEmpty
              iconType="indexOpen"
              onClick={[Function]}
            >
              Add data
            </EuiButtonEmpty>
          </EuiFlexItem>
          <EuiFlexItem
            className="kbnOverviewHeader__actionItem"
            grow={false}
          >
            <EuiButtonEmpty
              iconType="gear"
              onClick={[Function]}
            >
              Manage
            </EuiButtonEmpty>
          </EuiFlexItem>
          <EuiFlexItem
            className="kbnOverviewHeader__actionItem"
            grow={false}
          >
            <EuiButtonEmpty
              iconType="wrench"
              onClick={[Function]}
            >
              Dev tools
            </EuiButtonEmpty>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiFlexItem>
    </EuiFlexGroup>
  </div>
</header>
`;

exports[`PageHeader with dev tools 1`] = `
<header
  className="kbnOverviewHeader"
>
  <div
    className="kbnOverviewHeader__inner"
  >
    <EuiFlexGroup>
      <EuiFlexItem>
        <EuiFlexGroup
          gutterSize="m"
          responsive={false}
        >
          <EuiFlexItem
            grow={false}
          >
            <EuiIcon
              size="xxl"
              type="logoKibana"
            />
          </EuiFlexItem>
          <EuiFlexItem>
            <EuiTitle
              size="m"
            >
              <h1
                id="kbnOverviewHeader__title"
              >
                <FormattedMessage
                  defaultMessage="Kibana"
                  id="kibana.overview.header.title"
                  values={Object {}}
                />
              </h1>
            </EuiTitle>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiFlexItem>
      <EuiFlexItem
        grow={false}
      >
        <EuiFlexGroup
          className="kbnOverviewHeader__actions"
          responsive={false}
          wrap={true}
        >
          <EuiFlexItem
            className="kbnOverviewHeader__actionItem"
            grow={false}
          >
            <EuiButtonEmpty
              iconType="indexOpen"
              onClick={[Function]}
            >
              Add data
            </EuiButtonEmpty>
          </EuiFlexItem>
          <EuiFlexItem
            className="kbnOverviewHeader__actionItem"
            grow={false}
          >
            <EuiButtonEmpty
              iconType="wrench"
              onClick={[Function]}
            >
              Dev tools
            </EuiButtonEmpty>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiFlexItem>
    </EuiFlexGroup>
  </div>
</header>
`;

exports[`PageHeader with management 1`] = `
<header
  className="kbnOverviewHeader"
>
  <div
    className="kbnOverviewHeader__inner"
  >
    <EuiFlexGroup>
      <EuiFlexItem>
        <EuiFlexGroup
          gutterSize="m"
          responsive={false}
        >
          <EuiFlexItem
            grow={false}
          >
            <EuiIcon
              size="xxl"
              type="logoKibana"
            />
          </EuiFlexItem>
          <EuiFlexItem>
            <EuiTitle
              size="m"
            >
              <h1
                id="kbnOverviewHeader__title"
              >
                <FormattedMessage
                  defaultMessage="Kibana"
                  id="kibana.overview.header.title"
                  values={Object {}}
                />
              </h1>
            </EuiTitle>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiFlexItem>
      <EuiFlexItem
        grow={false}
      >
        <EuiFlexGroup
          className="kbnOverviewHeader__actions"
          responsive={false}
          wrap={true}
        >
          <EuiFlexItem
            className="kbnOverviewHeader__actionItem"
            grow={false}
          >
            <EuiButtonEmpty
              iconType="indexOpen"
              onClick={[Function]}
            >
              Add data
            </EuiButtonEmpty>
          </EuiFlexItem>
          <EuiFlexItem
            className="kbnOverviewHeader__actionItem"
            grow={false}
          >
            <EuiButtonEmpty
              iconType="gear"
              onClick={[Function]}
            >
              Manage
            </EuiButtonEmpty>
          </EuiFlexItem>
        </EuiFlexGroup>
      </EuiFlexItem>
    </EuiFlexGroup>
  </div>
</header>
`;
